### PR TITLE
Feature/pillar staking v2

### PR DESCRIPTION
**Pull request by poocart**
🚨 🚨 🚨 🚨 🚨  
- This does not include actual staking contract as it is said to be deployed later 
- Unstaking is not available and will be available sometime later (product team has dates)

🚨 🚨 🚨 🚨 🚨

## Description
- https://linear.app/pillarproject/issue/PRO-887/plr-staking-v2-block

## Motivation and Context
- https://linear.app/pillarproject/issue/PRO-887/plr-staking-v2-block

## How Has This Been Tested?

- Locally on mainnet.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by Yash-Chaubal — COMMENTED** [review]
Looking good! Just added a couple of comments

File: src/components/TransactionBlock/PlrStakingV2TransactionBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
+};
+
+const mapRouteToOption = (route: Route) => {
+  const [fistStep] = route.steps;
+  const serviceDetails = bridgeServiceIdToDetails[fistStep?.toolDetails?.key ?? 'lifi'];
+  return {
+    title: fistStep?.toolDetails?.name ?? serviceDetails?.title ?? 'LiFi',
+    value: route.id,
+    iconUrl: fistStep?.toolDetails?.logoURI ?? serviceDetails?.iconUrl,
+  };
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/components/TransactionBlock/PlrStakingV2TransactionBlock.tsx:125]: correct, fixed

File: src/providers/TransactionBuilderContextProvider.tsx [hunk 2]
```diff
@@ -295,7 +297,7 @@ const TransactionBuilderContextProvider = ({
 
   const [copiedAddress, setCopiedAddress] = useState(false);
   const [copiedAddressInterval, setCopiedAddressInterval] = useState<number | null>(null);
-  const [showWalletBlock, setShowWalletBlock] = useState(true);
+  const [showWalletBlock, setShowWalletBlock] = useState(hideWalletBlock ?? true);
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/providers/TransactionBuilderContextProvider.tsx]: done, ty

File: src/components/TransactionPreview/ActionPreview.tsx [hunk 3]
```diff
@@ -1212,6 +1212,114 @@ const ActionPreview = ({
...
+          <TransactionAction>
+            <Label>Route</Label>
+            <RouteOption route={swap.route} cost={cost} showActions />
+          </TransactionAction>
+        )}
+        {showGasAssetSelect && <GasTokenSelect crossChainAction={crossChainAction} />}
+        {showStatus && (
+          <TransactionStatus
+            crossChainAction={crossChainAction}
+            setIsTransactionDone={setIsTransactionDone ?? undefined}
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/components/TransactionPreview/ActionPreview.tsx:1334]: done, ty

File: src/utils/transaction.ts [hunk 4]
```diff
@@ -1115,6 +1116,173 @@ export const buildCrossChainAction = async (
...
+        isEstimating: false,
+        estimated: null,
+        useWeb3Provider: accountType === AccountTypes.Key,
+        multiCallData: transactionBlock?.multiCallData,
+      };
+
+      return { crossChainAction };
+    } catch (e) {
+      return { errorMessage: 'Failed to build swap transaction!' };
+    }
```
[the remark this answers is not in the document]
Comment (poocart) [reply src/utils/transaction.ts]: good catch (pun intended), added

**Review by Yash-Chaubal — APPROVED** [review]
Everything LGTM 🚀

**Pull request merged**